pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -101,13 +102,18 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(proxy *model.Proxy, push *mo
 		clusters = append(clusters, patcher.insertedClusters()...)
 	}
 
-	return cb.normalizeClusters(clusters)
+	for _, c := range clusters {
+		resources = append(resources, &discovery.Resource{Name: c.Name, Resource: util.MessageToAny(c)})
+	}
+	resources = cb.normalizeClusters(resources)
+	return resources
 }
 
-func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder, cp clusterPatcher) []*cluster.Cluster {
-	clusters := make([]*cluster.Cluster, 0)
+// buildOutboundClusters generates all outbound (including subsets) clusters for a given proxy.
+func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder, cp clusterPatcher) []*discovery.Resource {
+	resources := make([]*discovery.Resource, 0)
 	networkView := cb.proxy.GetNetworkView()
-
+	hit, miss := 0, 0
```

Comment:
Where are these used? Intended to emit metrics?

Reply:
Ah yeah, meant to return something like https://github.com/istio/istio/blob/cd4e73cd8fdca373d89ae1b8175a1b0c2d4d0658/pilot/pkg/xds/eds.go#L426 but forgot to finish it out.